### PR TITLE
Fixes and features - v1

### PR DESCRIPTION
**Pull request by silsanchez8**
## Modifications:
- React router
- Component states
- getting data from API
- Add simple test

### CODE REVIEW

File: users-react-app/src/pages/Layout.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import Header from '../components/Header/Header';
+
+function Layout(props){
+    return (
+        <React.Fragment>
```
Comment (cbernat) [inline users-react-app/src/pages/Layout.js]: This should work too and it´s less verbose 😄 
```suggestion
        <>
```

File: users-react-app/src/pages/NewUser.js [hunk 2]
```diff
@@ -3,45 +3,45 @@ import Card from '../components/Card/Card';
...
                 ...this.state.form,
-                [e.target.name] : e.target.value,
+                [e.target.name]: e.target.value,
             }
         })
     }
     render() {
         return (
-            <>
+            <React.Fragment>
```
Comment (cbernat) [inline users-react-app/src/pages/NewUser.js]:
```suggestion
            <>
```

File: users-react-app/src/pages/NewUser.js [hunk 3]
```diff
@@ -3,45 +3,45 @@ import Card from '../components/Card/Card';
...
-            </>
+            </React.Fragment>
         );
     }
 
 }
 
-export default UsersNew;
\ No newline at end of file
+export default NewUser;
```
Comment (cbernat) [inline users-react-app/src/pages/NewUser.js]: You have a missing new line at the end.

File: users-react-app/src/App.js [hunk 4]
```diff
@@ -1,16 +1,25 @@
...
 function App() {
   return (
-    <div className="App">
-       <Header></Header>
-      <UsersNew />
-    </div>
+    <Layout>
+      <BrowserRouter>
+      <Switch>
+        <Route exact path='/' component={WelcomeMessage} />
```
Comment (cbernat) [inline users-react-app/src/App.js]: How about adding a test for each route?

File: users-react-app/src/pages/Layout.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import Header from '../components/Header/Header';
+
+function Layout(props){
```
Comment (cbernat) [inline users-react-app/src/pages/Layout.js]: How about using an arrow function for this functional component? Like this:
```suggestion
export const Layout = (props) => {
```
And be done with the export in the same line too 😄

File: users-react-app/src/pages/NotFound.js [hunk 6]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+
+function NotFound() {
+    return (
+        <h1> 404 - Not Found</h1>
+    )
+}
+export default NotFound;
```
Comment (cbernat) [inline users-react-app/src/pages/NotFound.js]: Missing new line at the end of the file.

File: users-react-app/src/pages/Users.js [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+                        ); 
+                    })}
+
+                </ul>
+            </div>
+            </React.Fragment>
+        );
+    }
+};
+export default Users;
```
Comment (cbernat) [inline users-react-app/src/pages/Users.js]: Missing new line at the end of the file.

File: users-react-app/src/pages/WelcomeMessage.js [hunk 8]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+
+function WelcomeMessage(){
+    return (
+        <h1>Welcome to ABM USERS</h1>
+    );
+}
+export default WelcomeMessage;
```
Comment (cbernat) [inline users-react-app/src/pages/WelcomeMessage.js]: Missing new line at the end of the file.

File: users-react-app/src/App.js [hunk 9]
```diff
@@ -1,16 +1,25 @@
...
 
 function App() {
   return (
-    <div className="App">
-       <Header></Header>
-      <UsersNew />
-    </div>
+    <Layout>
+      <BrowserRouter>
+      <Switch>
```
Comment (cbernat) [inline users-react-app/src/App.js]: Watch out for the tabulation here! 
```suggestion
      <BrowserRouter>
        <Switch>
```

File: users-react-app/src/App.js (hunk 4 above)
Comment (silsanchez8) [reply users-react-app/src/App.js]: I've applied a test for welcomeMessage.js, I'll add a test for Routes as soon as posible.

File: users-react-app/src/pages/Layout.js (hunk 5 above)
Comment (silsanchez8) [reply users-react-app/src/pages/Layout.js]: I resolved it : 
`const Layout = (props) => {
  return (
    <>
      <Header />
      {props.children}
    </>
  );
};
export default Layout;`

How can you apply export in the same line? I've tried this way but it does not work:

`export const Layout = (props) => {
  return (
    <>
      <Header />
      {props.children}
    </>
  );
};
 `

Console throw this error:

image

Comment (cbernat) [reply users-react-app/src/pages/Layout.js]: You don`t need to add al the funcion in the same line 👀

File: users-react-app/src/components/UserForm/UserForm.js [hunk 10]
```diff
@@ -1,68 +1,97 @@
...
-
+import React from "react";
+import "../../styles/user-form.css";
 
 class UserForm extends React.Component {
-    
-    handleChange = (e) => {
-        /*console.log({
+  handleChange = (e) => {
+    /*console.log({
```
Comment (cbernat) [inline users-react-app/src/components/UserForm/UserForm.js]: Why do you have commented code here? Is it ok?

**Comment by cbernat** [comment]
Some small comment and it's ready to go! 🚀 Good work!

**Pull request merged**